extensions/api/data-management/contractnegotiation/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/contractnegotiation/service/ContractNegotiationServiceImpl.java

```diff
@@ -0,0 +1,117 @@
+import org.eclipse.dataspaceconnector.spi.types.domain.contract.negotiation.ContractOfferRequest;
+import org.eclipse.dataspaceconnector.spi.types.domain.contract.offer.ContractDefinition;
+
+import java.util.Collection;
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static java.lang.String.format;
+import static java.util.stream.Collectors.toList;
+
+public class ContractNegotiationServiceImpl implements ContractNegotiationService {
+
+    private final ContractNegotiationStore store;
+    private final ConsumerContractNegotiationManager manager;
+    private final TransactionContext transactionContext;
+
+    public ContractNegotiationServiceImpl(ContractNegotiationStore store, ConsumerContractNegotiationManager manager, TransactionContext transactionContext) {
+        this.store = store;
+        this.manager = manager;
+        this.transactionContext = transactionContext;
+    }
+
+    @Override
+    public ContractNegotiation findbyId(String contractNegotiationId) {
+        return store.find(contractNegotiationId);
+    }
+
+    @Override
+    public Collection<ContractNegotiation> query(QuerySpec query) {
+        return store.queryNegotiations(query).collect(toList());
+    }
+
+    @Override
+    public String getState(String negotiationId) {
+        return Optional.of(negotiationId)
+                .map(this::findbyId)
+                .map(ContractNegotiation::getState)
+                .map(ContractNegotiationStates::from)
+                .map(Enum::name)
+                .orElse(null);
+    }
+
+    @Override
+    public ContractAgreement getForNegotiation(String negotiationId) {
+        return store.findContractAgreement(negotiationId);
+    }
+
+    @Override
+    public ContractNegotiation initiateNegotiation(ContractOfferRequest request) {
+        var result = manager.initiate(request);
+        return result.getContent();
+    }
+
+    @Override
+    public ServiceResult<ContractNegotiation> cancel(String negotiationId) {
+        var result = new AtomicReference<ServiceResult<ContractNegotiation>>();
+
+        transactionContext.execute(() -> {
+            var negotiation = store.find(negotiationId);
+            if (negotiation == null) {
+                result.set(ServiceResult.notFound(format("ContractNegotiation %s does not exist", negotiationId)));
+                return;
+            }
+            manager.enqueueCommand(new CancelNegotiationCommand(negotiationId));
+            result.set(ServiceResult.success(negotiation));
+        });
+
+        return result.get();
+    }
+
+    @Override
+    public ServiceResult<ContractNegotiation> decline(String negotiationId) {
+        var result = new AtomicReference<ServiceResult<ContractNegotiation>>();
+
+        transactionContext.execute(() -> {
+            try {
+                var declineResult = manager.declined(ClaimToken.Builder.newInstance().build(), negotiationId);
```

Comment:
Not sure to understand why here we are not using a command to decline the negotiation like in the `cancel` method

Reply:
it was like this since the first version of the `ContractNegotiation` interface. Please address this to another issue